### ISSUE TITLE
Make methods return &(mut) self where applicable

### ISSUE THREAD

**Issue by lukaslueg** [atroche/rust-headless-chrome#14]
A lot of methods currently return `Result<(), Error>`. More comfort could be provided by returning `Result<&Self, Error>` instead. This causes no change if the `Ok` is not of interest. Yet long chains like

```rust
let tab = browser.wait_for_initial_tab()?;
tab.navigate_to("https://www.wikipedia.org")?;
tab.wait_for_element(r#"input#searchInput"#)?.click()?;
tab.type_str("WebKit")?;
tab.press_key("Enter")?;
```

could be written as

```rust
let tab = browser.wait_for_initial_tab()?;
tab.navigate_to("https://www.wikipedia.org")?
   .wait_for_element(r#"input#searchInput"#)?.click()?
   .type_str("WebKit")?
   .press_key("Enter")?;
```

**Comment by atroche** [comment]
Yeah, I'm into that. Are you also proposing putting the `type_str` and `press_key` methods on `Element` as well?

**Comment by lukaslueg** [comment]
no, sorry, this was a cooked-up example and I did not check if the types actually chain.

**Comment by lukaslueg** [comment]
closed in 6f337722612ccd20646f80a01070d38625b8b009
